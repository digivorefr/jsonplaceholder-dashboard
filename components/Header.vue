<template>
  <header class="header">
    <div class="container header__wrapper">
      <nav class="header__nav">
        <nuxt-link to="/">
          All users
        </nuxt-link>
      </nav>
      <input
        id="theme"
        type="checkbox"
        name="theme"
        style="opacity:0; position:absolute; visibility: hidden; z-index:-1; pointer-events: none;"
        :checked="theme === 'dark-mode'"
        @change="onThemeChange"
      >
      <label
        id="theme-switcher"
        for="theme"
      >
        <div class="icon icon--sun" />
        <div class="icon icon--moon" />
      </label>
      <div class="profile">
        <p class="profile__name">
          Marty McFly
        </p>
        <a
          class="profile__picture"
          href="https://youtu.be/fCjsUxbNmIs?t=102"
          target="_blank"
        >
          <img
            src="~/static/images/profile.jpg"
            alt="Profile picture"
          >
        </a>
      </div>
    </div>
  </header>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  name: 'MainHeader',
  data() {
    return {
      route: this.$route.path,
      theme: this.$theme
    }
  },
  methods: {
    onThemeChange(event: any) {
      if (event?.target?.checked !== null && event?.target?.checked === true) {
        this.$changeTheme('dark-mode')
      } else {
        this.$changeTheme('light-mode')
      }
    }
  }
})
</script>
