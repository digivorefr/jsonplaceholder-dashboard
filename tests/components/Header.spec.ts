import { shallowMount } from '@vue/test-utils'
import Header from '@/components/Header.vue'

describe('Header', () => {
  test('renders correctly', () => {
    const wrapper = shallowMount(Header, {
      mocks: {
        $route: {
          path: '/'
        }
      }
    })
    expect(wrapper.vm).toBeTruthy()
    expect(wrapper.exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
    wrapper.destroy()
  })

  test('switch theme', async () => {
    const changeTheme = jest.fn()

    const wrapper = shallowMount(Header, {
      mocks: {
        $route: {
          path: '/'
        },
        $theme: 'light-mode',
        $changeTheme: changeTheme
      }
    })

    expect(wrapper.vm.$data.theme).toBe('light-mode')
    const switcher = wrapper.find('#theme')
    await switcher.setChecked()
    await switcher.setChecked(false)
    expect(changeTheme).toHaveBeenNthCalledWith(1, 'dark-mode')
    expect(changeTheme).toHaveBeenNthCalledWith(2, 'light-mode')
    wrapper.destroy()
  })
})
