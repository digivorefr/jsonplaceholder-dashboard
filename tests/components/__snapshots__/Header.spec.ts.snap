// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  class="header"
>
  <div
    class="container header__wrapper"
  >
    <nav
      class="header__nav"
    >
      <a
        to="/"
      >
        
        All users
      
      </a>
    </nav>
     
    <input
      id="theme"
      name="theme"
      style="opacity: 0; position: absolute; visibility: hidden; z-index: -1; pointer-events: none;"
      type="checkbox"
    />
     
    <label
      for="theme"
      id="theme-switcher"
    >
      <div
        class="icon icon--sun"
      />
       
      <div
        class="icon icon--moon"
      />
    </label>
     
    <div
      class="profile"
    >
      <p
        class="profile__name"
      >
        
        Marty McFly
      
      </p>
       
      <a
        class="profile__picture"
        href="https://youtu.be/fCjsUxbNmIs?t=102"
        target="_blank"
      >
        <img
          alt="Profile picture"
          src="~/static/images/profile.jpg"
        />
      </a>
    </div>
  </div>
</header>
`;
